### PR TITLE
activating checkboxes for data-source-buckets #2107

### PR DESCRIPTION
**Pull request by Angamanga**
This pull request makes the following changes:
- activates the checkboxes for data buckets in map-view

Testing checklist:
- [ ] go to map-view
- [ ] checkboxes for data-sources should be checked
- [ ] deselect one checkbox
- [ ] posts from that data-source should be removed from the map
- [ ] the filter should not be selected in the filter- dropdow nor be visible as a bug-icon in the top of the dropdown
- [ ] select the checkbox again
- [ ] post should appear
- [ ] the filter should be checked in the filter-dropdown 

Fixes ushahidi/platform#2107 .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by crcommons** [comment]
Just tested locally...

- only Twitter and SMS data sources were listed; Web and Email were missing
- The data sources have the styling/css indicating they are selected, but they are not actually checked 
screen shot 2017-11-02 at 10 45 54 am
- if I deselect a data source, it adds a checkmark and then quickly removes it; it DOES remove them from the map though, so that's good!

**Comment by Angamanga** [comment]
> only Twitter and SMS data sources were listed; Web and Email were missing
1. Web is not listed as a datasource, as of the original design, but it does make sense to have them there. That is another issue though. @justinscherer @jess any comments on making "Web" as a separate data-source-bucket?

2. Where were you testing? The only datasources visible is the ones enabled or have data present in the database.

> The data sources have the styling/css indicating they are selected, but they are not actually checked
>- if I deselect a data source, it adds a checkmark and then quickly removes it
Fixed :)

**Comment by crcommons** [comment]
Changes look good. As long as we don't need to add Web in, then it's good to go!

File: test/unit/main/post/views/filter-by-datasource.spec.js [hunk 1]
```diff
@@ -90,5 +90,12 @@ describe('filter by datasource directive', function () {
             isolateScope.showOnlyIncoming('SMS');
             expect($location.path()).toEqual('/views/data');
         });
+        it('should toggle the filters based on if selected filter is activated or not', function () {
+            console.info(isolateScope.filters.source);
```
Comment (crcommons) [inline test/unit/main/post/views/filter-by-datasource.spec.js]: Should you remove the console.info here?

Comment (Angamanga) [reply test/unit/main/post/views/filter-by-datasource.spec.js]: Hehe, yes I should, fixed now...

**Pull request merged**